worker/cppworker/worker/OCCChild.cpp

```diff
@@ -5704,6 +5718,21 @@ uint OCCChild::compute_scuttle_id(unsigned long long _shardkey_val)
 	return scuttle_id;
 }
 
+uint OCCChild::compute_scuttle_id(std::string _shardkey_str_val)
+{
+	uint scuttle_id = 0;
+	switch(m_sharding_algo)
+	{
+		case MOD_ONLY:
```

Comment:
MOD_ONLY for string shard key isn't correct. If we enable any use cases later, we will run into the same issue of scuttle mismatch. We could either correct it or not support MOD_ONLY for now.

Reply:
The implementation matches with what we have in mux. 
https://github.com/paypal/hera/blob/master/lib/coordinatorsharding.go#L86

Reply:
added a comment for clarity as suggested